src/sentry/workflow_engine/models/incident_groupopenperiod.py

```diff
@@ -32,3 +38,172 @@ class Meta:
                 name="inc_id_inc_identifier_together",
             )
         ]
+
+    @classmethod
+    def create_from_occurrence(cls, occurrence, group, open_period):
+        """
+        Creates an IncidentGroupOpenPeriod relationship from an issue occurrence.
+        This method handles the case where the incident might not exist yet.
+
+        Args:
+            occurrence: The IssueOccurrence that triggered the group creation
+            group: The Group that was created
+            open_period: The GroupOpenPeriod for the group
+        """
+        try:
+            # Extract alert_id from evidence_data
+            alert_id = occurrence.evidence_data.get("alert_id")
+            if not alert_id:
+                logger.warning(
+                    "No alert_id found in evidence_data for metric issue",
```

Comment:
hmm - how can we tie this to the new system only? that way when we come back to remove the open period / incident stuff we'll know this is created as expected?